### ISSUE TITLE
Exclude private repositories by default

### ISSUE THREAD

**Issue by atanisoft** [opulo-inc/hardware-github-viewer#2]
Private repositories should not include the links to view the model since the external sites do not have read access to private repositories. Embedding the KiCanvas code might work but it would need to use an authenticated call to retrieve the KiCad files from the private repository.

**Comment by sphawes** [comment]
oh, damn that's a great thought! weird edge case. shouldnt be too hard to check if the repo is private, and effectively disable the extension. the auth might be weird, not sure how we'd go about doing that.

**Comment by atanisoft** [comment]
Yeah, it's definitely an edge case. I'm not sure how auth might be possible outside of using something like `fetch` to retrieve the file into a local variable and pass the content along to the viewer as maybe base64 encoded (or for direct embedding pass the var itself).
